iac-extensions/kubernetes/src/test/resources/checks/CpuRequestCheck/cpu_request_kind_template.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
+kind: ${type}
+metadata:
+  name: example
+  labels:
+    app: web
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: web
+  template:
+    metadata:
+      labels:
+        app: web
+    spec:
+      containers:
+        - name: noncompliant_without_cpu_request # Noncompliant {{Specify a CPU request for this container.}}
+        # ^^^^
```

Comment:
It could be addressed, but I'm not sure if the tradeoff is worth it.
Currently we're raising on the key of the first child element of the container. (not specifically searching for `name` here, as it doesn't have to be the first element.

If we change it to raise on the whole first child element, we would potentially raise on multiple lines, depending which element we encounter at first.
In this example we would raise on everything starting from the second line.
```
  containers:
    - volumeMounts:
          # name must match the volume name below
        - name: mysql-persistent-storage
          # mount path within the container
          mountPath: /var/lib/mysql
```

We could make this logic more sophisticated, to only raise on the whole child if the value is a string, and on the key if the value is something else.
But that complicates the check a lot, and I'm not sure it's worth it.

WDYT?

Reply:
Indeed, I don't think it is worth it, thanks for the explanation 👍